docs/src/pages/components/autocomplete/Virtualize.tsx

```diff
@@ -1,58 +1,70 @@
+import Autocomplete, { RenderGroupParams } from '@material-ui/lab/Autocomplete';
 import useMediaQuery from '@material-ui/core/useMediaQuery';
+import ListSubheader from '@material-ui/core/ListSubheader';
 import { useTheme, makeStyles } from '@material-ui/core/styles';
-import { FixedSizeList, ListChildComponentProps } from 'react-window';
+import { VariableSizeList, ListChildComponentProps } from 'react-window';
+import { Typography } from '@material-ui/core';
 
 function renderRow(props: ListChildComponentProps) {
   const { data, index, style } = props;
-
   return React.cloneElement(data[index], {
     style: {
-      overflow: 'hidden',
-      textOverflow: 'ellipsis',
-      whiteSpace: 'nowrap',
-      display: 'block',
       ...style,
+      top: (style.top as number) + 8,
     },
   });
 }
 
 // Adapter for react-window
-const ListboxComponent = React.forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLElement>>(
-  function ListboxComponent(props, ref) {
-    const { children, ...other } = props;
-    const theme = useTheme();
-    const smUp = useMediaQuery(theme.breakpoints.up('sm'));
-    const itemCount = Array.isArray(children) ? children.length : 0;
-    const itemSize = smUp ? 36 : 48;
+const ListboxComponent = React.forwardRef<HTMLDivElement>(function ListboxComponent(props, ref) {
+  const { children, ...other } = props;
+  const itemData = React.Children.toArray(children);
+  const theme = useTheme();
+  const smUp = useMediaQuery(theme.breakpoints.up('sm'), { noSsr: true });
+  const itemCount = itemData.length;
+  const itemSize = smUp ? 36 : 48;
 
-    const outerElementType = React.useMemo(() => {
-      return React.forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLDivElement>>(
-        (props2, ref2) => <div ref={ref2} {...props2} {...other} />,
-      );
-    }, []); // eslint-disable-line react-hooks/exhaustive-deps
+  const getChildSize = (child: React.ReactNode) => {
+    if (React.isValidElement(child) && child.type === ListSubheader) {
+      return 48;
+    }
 
-    return (
-      <div ref={ref}>
-        <FixedSizeList
-          style={{ padding: 0, height: Math.min(8, itemCount) * itemSize, maxHeight: 'auto' }}
-          itemData={children}
-          height={250}
-          width="100%"
-          outerElementType={outerElementType}
-          innerElementType="ul"
-          itemSize={itemSize}
-          overscanCount={5}
-          itemCount={itemCount}
-        >
-          {renderRow}
-        </FixedSizeList>
-      </div>
-    );
-  },
-);
+    return itemSize;
+  };
+
+  const getHeight = () => {
+    if (itemCount > 8) {
+      return 8 * itemSize;
+    }
+    return itemData.map(getChildSize).reduce((a, b) => a + b, 0);
+  };
+
+  const outerElementType = React.useMemo(() => {
+    return React.forwardRef<HTMLDivElement>((props2, ref2) => (
+      <div ref={ref2} {...props2} {...other} />
+    ));
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
@oliviertassinari I'm not sure I fully understand the implications of this. Wouldn't this mean that only the props of the first render get passed through?

Reply:
Yes, it's no ideal. I couldn't find another approach to apply props on the outer element.
https://github.com/bvaughn/react-window/issues/242

Reply:
Did you try with a context as per https://github.com/bvaughn/react-window/issues/242#issuecomment-493877443 ?

Reply:
No, I haven't tried. In the current implementation, we don't need to handle an update of the props.

Reply:
Maybe we will in the future, people will have to come back to the demo to check the new logic.

Reply:
The component being in the lab, people can expect breaking changes.

Reply:
Quickly tried in my project with

```js

const outerElementPropsContext = React.createContext({});

const OuterElementType = React.forwardRef((props, ref) => {
  const outerProps = React.useContext(outerElementPropsContext);
  return <div ref={ref} {...props} {...outerProps} />;
});

// ...

  return (
    <div ref={ref}>
      <outerElementPropsContext.Provider value={other}>
        <VariableSizeList
// ...
        </VariableSizeList>
      </outerElementPropsContext.Provider>
    </div>
  );
```

And at first sight that seems to work. I can update the PR if you want?

Reply:
This looks even better :)

Reply:
Do you want to apply the changes?

Reply:
I'm sorry, I'm at work. I'll try to look into it in a couple of hours when I'm home.

Reply:
@oliviertassinari Done

Reply:
Thanks for the fast response time!